### PR TITLE
enable x-platform analytics

### PR DESCRIPTION
**Pull request by ecotopian**

### CODE REVIEW

File: Client.xcodeproj/project.pbxproj [hunk 1]
```diff
@@ -12822,8 +12822,8 @@
 			isa = XCRemoteSwiftPackageReference;
 			repositoryURL = "https://github.com/ecosia/ios-core.git";
 			requirement = {
-				kind = exactVersion;
-				version = 4.3.0;
```
Comment (ecotopian) [inline Client.xcodeproj/project.pbxproj:12826]: will change to tag once PR in Core is merged

File: Client/Frontend/Browser/BrowserViewController/BrowserViewController+WebViewDelegates.swift [hunk 2]
```diff
@@ -695,6 +696,12 @@ extension BrowserViewController: WKNavigationDelegate {
                     }
                 }
             }
+
+            if !User.shared.searched, let url = webView.url,
+               url.isEcosia, url.path.hasPrefix("/search") {
+
+                User.shared.searched = true
```
Comment (ecotopian) [inline Client/Frontend/Browser/BrowserViewController/BrowserViewController+WebViewDelegates.swift:703]: prevents appending *firstrun* and *addon* params for all searches

File: Client/Frontend/Browser/BrowserViewController/BrowserViewController+WebViewDelegates.swift [hunk 3]
```diff
@@ -695,6 +696,12 @@ extension BrowserViewController: WKNavigationDelegate {
                     }
                 }
             }
+
+            if !User.shared.searched, let url = webView.url,
+               url.isEcosia, url.path.hasPrefix("/search") {
```
Comment (vauxhall) [inline Client/Frontend/Browser/BrowserViewController/BrowserViewController+WebViewDelegates.swift:701]: In the feedback PR to your Core PR I added a constant for search, so we don't type `"/search"` in many places.

**Review by vauxhall — APPROVED** [review]
I would recommend to first have a look at the feedback PR on Core and then update if necessary, otherwise it looks great 👍

File: Client/Frontend/Browser/Tab.swift [hunk 4]
```diff
@@ -456,7 +456,7 @@ class Tab: NSObject {
             if !isPrivate, let cookie = browserViewController?.tabManager.cookie {
                 var request = request
                 if let url = request.url, url.isEcosia {
-                    request.url = url.addQuery(name: "_sp", value: User.shared.analyticsId.uuidString)
+                    request.url = url.ecosified()
```
Comment (vauxhall) [inline Client/Frontend/Browser/Tab.swift:459]: 👍

**Review by vauxhall — APPROVED** [review]

**Pull request merged**